src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -114,5 +127,45 @@ private bool ShouldRestoreOnBuild
                 return packageRestoreConsent.IsAutomatic;
             }
         }
+
+        #region IVsUpdateSolutionEvents4
+
+        public void UpdateSolution_QueryDelayFirstUpdateAction(out int pfDelay)
+        {
+            // check if NuGet lock is already acquired by some other NuGet operation
+            if (LockService.IsLockHeld)
+            {
+                // delay build by 1000ms and let it try again after that.
+                pfDelay = 1000;
```

Comment:
Please change this to just assign `1`. You don't get to dictate the interval. It's just zero or non-zero. And so while setting `1000` is the same as `1` functionally, it gives the impression to the code reader that you get to set the length, and that is not the case.
I would also avoid specifying in your code comment that the delay is 1 second long since that is an implementation detail that can change.